libsolidity/parsing/Parser.cpp

```diff
@@ -159,7 +159,13 @@ ASTPointer<SourceUnit> Parser::parse(CharStream& _charStream)
 void Parser::parsePragmaVersion(SourceLocation const& _location, vector<Token> const& _tokens, vector<string> const& _literals)
 {
 	SemVerMatchExpressionParser parser(_tokens, _literals);
-	auto matchExpression = parser.parse();
+	auto matchExpressionOrError = parser.parse();
+	if (std::holds_alternative<SemVerError>(matchExpressionOrError))
+	{
+		auto matchExpressionError = std::get<SemVerError>(matchExpressionOrError);
+		BOOST_THROW_EXCEPTION(matchExpressionError);
+	}
+	auto matchExpression = std::get<std::optional<SemVerMatchExpression>>(matchExpressionOrError);
```

Comment:
Is there still a need for the `optional`? Isn't it either an error or a valid match expression?

Reply:
I think you're right because of the way the match expression is handled after. I just wasn't sure whether `m_tokens.empty()` which would return a `nullopt` is a valid and handled return type (Line 163 of SemVerHandler.cpp)

I will alter it to just an error or valid match expression.